Source/DafnyCore/Backends/SinglePassCodeGenerator/SinglePassCodeGenerator.cs

```diff
@@ -2129,6 +2132,7 @@ protected class NullClassWriter : IClassWriter {
       if (c is not TraitDecl || TraitRepeatsInheritedDeclarations) {
         thisContext = c;
         foreach (var member in inheritedMembers.Select(memberx => (memberx as Function)?.ByMethodDecl ?? memberx)) {
+          enclosingDeclaration = member;
```

Comment:
Would be nice to have this value in a `CompilationContext` parameter this is passed down, instead of state on the generator class, but we can change that another time.